lib/versioning/pep440/index.spec.ts

```diff
@@ -84,71 +84,84 @@ describe('versioning/pep440/index', () => {
+    ${'>=2.0.0'}            | ${'replace'}     | ${'1.0.0'}     | ${'1.2.3'}   | ${'>=1.2.3'}
+    ${'>=2.0.0'}            | ${'pin'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'==1.2.3'}
+    ${'~=1.1.0, !=1.1.1'}   | ${'bump'}        | ${'1.0.0'}     | ${'1.2.3'}   | ${'~=1.2.3, !=1.1.1'}
+    ${'~=1.1.0, !=1.1.1'}   | ${'replace'}     | ${'1.0.0'}     | ${'1.2.3'}   | ${'~=1.2.3, !=1.1.1'}
+    ${'~=1.1.0, !=1.1.1'}   | ${'pin'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'==1.2.3'}
+    ${'~=1.1.0,!=1.1.1'}    | ${'bump'}        | ${'1.0.0'}     | ${'1.2.3'}   | ${'~=1.2.3,!=1.1.1'}
+    ${'~=1.1.0,!=1.1.1'}    | ${'replace'}     | ${'1.0.0'}     | ${'1.2.3'}   | ${'~=1.2.3,!=1.1.1'}
+    ${'~=1.1.0,!=1.1.1'}    | ${'pin'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'==1.2.3'}
+    ${' '}                  | ${'bump'}        | ${'1.0.0'}     | ${'1.2.3'}   | ${' '}
+    ${' '}                  | ${'replace'}     | ${'1.0.0'}     | ${'1.2.3'}   | ${' '}
+    ${' '}                  | ${'pin'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'==1.2.3'}
+    ${'invalid'}            | ${'bump'}        | ${'1.0.0'}     | ${'1.2.3'}   | ${null}
+    ${'invalid'}            | ${'replace'}     | ${'1.0.0'}     | ${'1.2.3'}   | ${null}
+    ${'invalid'}            | ${'pin'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'==1.2.3'}
+    ${'===1.0.3'}           | ${'bump'}        | ${'1.0.0'}     | ${'1.2.3'}   | ${null}
+    ${'===1.0.3'}           | ${'replace'}     | ${'1.0.0'}     | ${'1.2.3'}   | ${null}
+    ${'===1.0.3'}           | ${'pin'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'==1.2.3'}
+    ${'!=1.2.3'}            | ${'bump'}        | ${'1.0.0'}     | ${'1.2.3'}   | ${null}
+    ${'!=1.2.3'}            | ${'replace'}     | ${'1.0.0'}     | ${'1.2.3'}   | ${null}
+    ${'!=1.2.3'}            | ${'pin'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'==1.2.3'}
+    ${'~=1.1.0,!=1.1.1'}    | ${'unsupported'} | ${'1.0.0'}     | ${'1.2.3'}   | ${'~=1.2.3,!=1.1.1'}
+    ${'>=19.12.2,<20.13.9'} | ${'replace'}     | ${'19.12.2'}   | ${'21.3.1'}  | ${'>=21.3.1,<22.0.0'}
+    ${'>=19.12.2,<19.13.9'} | ${'replace'}     | ${'19.12.2'}   | ${'20.3.1'}  | ${'>=20.3.1,<20.4.0'}
+    ${'>=19.12.2,<19.13.0'} | ${'replace'}     | ${'19.12.2'}   | ${'20.3.1'}  | ${'>=20.3.1,<20.4.0'}
+    ${'>=19.12.2,<19.13.0'} | ${'replace'}     | ${'19.12.2'}   | ${'20.3.0'}  | ${'>=20.3.0,<20.4.0'}
+    ${'>=19.12.2,<19.13.0'} | ${'replace'}     | ${'19.12.2'}   | ${'19.13.1'} | ${'>=19.13.1,<19.14.0'}
+    ${'>=19.12.2,<19.13.0'} | ${'replace'}     | ${'19.12.2'}   | ${'19.13.0'} | ${'>=19.13.0,<19.14.0'}
+    ${'>=19.12.2,<19.13.0'} | ${'auto'}        | ${'19.12.2'}   | ${'19.13.0'} | ${'>=19.13.0,<19.14.0'}
+    ${'>=19.12.2,<20.13.9'} | ${'widen'}       | ${'19.12.2'}   | ${'21.3.1'}  | ${'>=19.12.2,<22.0.0'}
+    ${'>=19.12.2,<19.13.9'} | ${'widen'}       | ${'19.12.2'}   | ${'20.3.1'}  | ${'>=19.12.2,<20.4.0'}
```

Comment:
```suggestion
    ${'>=19.12.2,<20.13.9'} | ${'widen'}       | ${'19.12.2'}   | ${'21.3.1'}  | ${'>=19.12.2,<21.3.2'}
    ${'>=19.12.2,<19.13.9'} | ${'widen'}       | ${'19.12.2'}   | ${'20.3.1'}  | ${'>=19.12.2,<20.3.2'}
```

I suggest to match the "patch precision" of the upper bound here